### ISSUE TITLE
Feature request: allow getting IDs of related resources

### ISSUE THREAD

**Issue by pkopac** [recurly/recurly-client-ruby#377]
Hi! I'm using this library for integration in ChartMogul. One thing that I had to solve recently was optimization of number of requests. Eg. if I fetch all the invoices, I don't need to fetch singular ones again, therefore I need just the ID from a relation to look in the local cache. The library doesn't provide a straightforward way to achieve this and `invoice.original_invoice.uuid` results in a GET request. I've come up with the following, but it would be great, if the library included some method like `invoice.original_invoice_uuid`. Similar situation is with other relations. Thank you.
```
      # Returns ID without actually fetching the linked object.
      def get_id_from_links(obj, what)
        return nil if obj.blank?
        url = obj.links[what]
        return nil if url.blank?
        URI.parse(url[:href]).path.split('/')[-1]
      end
```

**Comment by ntippie** [comment]
I ran into a similar use case today. I wanted to see if an account had billing info, but discovered that calling `Recurly::Account#billing_info` makes a network call.

**Comment by bhelx** [comment]
This is a feature that is requested often. I don't like that the library opaquely makes HTTP requests for you in general, but we have to live with that for now. I'm not sure why a solution was not provided from the start for this problem. I'd like to provide a solution but I'm concerned that splitting and taking the last parameter won't work for all cases. Let me investigate this a little more before deciding what to do.

@pkopac are you monkey patching that into one of our classes? The Resource class perhaps? How do you use it?

**Comment by pkopac** [comment]
@bhelx, thank you for the response. Fortunately, no, I'm not monkey-patching this, although I had to other things :) We just use it as shown - a utility class method for getting around the problem wherever we detected repetitive GET calls. It's very important due to rate-limiting and efficiency.
It works with the cases we needed (original_invoice, plan, subscription). It doesn't work for eg. `invoice/:uuid/subscriptions` or `invoice/:uuid/original_invoices`, for apparent reasons.
